xbmc/cores/RetroPlayer/rendering/VideoRenderers/RPWinRenderer.cpp

```diff
@@ -205,12 +220,42 @@ bool CRPWinRenderer::UploadTexture()
 
 void CRPWinRenderer::Render(CD3DTexture *target)
 {
-  if (m_outputShader)
+  // Are we using video shaders?
+  if (m_useShaderPreset)
   {
-    m_outputShader->Render(*m_intermediateTarget, m_sourceWidth, m_sourceHeight,
-      m_sourceRect, m_rotatedDestCoords, target,
-      g_Windowing.UseLimitedColor() ? 1 : 0, 0.5f, 0.5f);
+    // select destination rectangle
+    CPoint destPoints[4];
+    if (m_renderOrientation)
+    {
+      for (size_t i = 0; i < 4; i++)
+        destPoints[i] = m_rotatedDestCoords[i];
+    }
+    else
+    {
+      CRect destRect = g_graphicsContext.StereoCorrection(m_destRect);
```

Comment:
Can you move some globals out of this class? Calling globals reduces the modularity of code.

Reply:
This is copied and pasted from WinRenderer.
How would I move it out of this class? Pass it as an arg from RPRenderManager to RenderUpdate -> Render? Could do the same thing for `g_Windowing.GetBackBuffer()` below but I'm not sure if it would make sense for the other renderers too.

Reply:
Any global depending on DX shouldn't be moved to the base class, so those can stay. geometry functions should definitely be moved up the class hierarchy

Reply:
Of course, it just seemed weird that the existing Kodi code wasn't written this way.